### ISSUE TITLE
On error of a video format, video-js does not fallback on other possible sources

### ISSUE THREAD

**Issue by cnico** [videojs/video.js#2133]
I have used the following simplest page of video-js : 
```
<body>
	<video id="example_video_1" class="video-js vjs-default-skin"
	  controls preload="auto" width="640" height="264"
	  poster="http://video-js.zencoder.com/oceans-clip.png"
	  data-setup=''>
		 <source src="./Velo.mp4" type='video/mp4' />
		 <source src="./Velo.webm" type='video/webm' />
		 <source src="./Velo.ogv" type='video/ogg' />
	 <p class="vjs-no-js">To view this video please enable JavaScript, and consider upgrading to a web browser that <a href="http://videojs.com/html5-video-support/" target="_blank">supports HTML5 video</a></p>
	</video>
</body>
```

In my videos, the first one is a mp4 but it does not contains a H.264 encoded video (it is a mistake of mine to have encoded it with another codec and it took me a long time to discover this mistake) and the webm video is correctly encoded.
When I play the video, I got en error "A network error caused the video download to fail part-way.".

If i invert the sources to put the webm first and mp4 second, the video plays well.

So I think, video-js should handle such situation of a video whose format is recognized by the browser but can not be displayed correctly in order to try automatically to play other video sources.

**Comment by mmcc** [comment]
I understand where you're coming from on this one, but I don't think I like this approach. I don't like the idea of the player blindly trying other sources, especially in your case since the error being thrown is network related.

I'm going to close this since I don't think this is something we're going to implement, but we can definitely continue the conversation in the comments. If something changes we can reopen the ticket.

**Comment by cnico** [comment]
Actually, the error message seems to be browser dependent since it is not the same message with other browsers or other platforms (mobile for example). 

If you think it is not a good design to implement it in video-js directly could you help me implementing it in my website ? It seems easy to catch the error but how can I switch the source in the list of the given source files and restart the video ?

The root cause problem is that probably that my video is not correctly encoded but I tried a lot of encoding options and encoding tools without success of portability throught most of browsers or devices.
Do you have one good documentation of how to encode videos for best video-js and most browsers compatibility ?

Thanks for your help.

**Comment by mmcc** [comment]
My apologies, I think I might have been a little hasty with my response the first time around. Especially in light of the native video element handling this case how you're expecting it to, I think we can consider the current behavior as a bug. I'm going to leave this issue closed, but only so we can consolidate in #869

**Comment by mmcc** [comment]
Also, in reference to the video encoding question, for one-off, short videos I generally use [Handbrake](http://handbrake.fr). The GUI is pretty easy to use and the presets are reasonable (make sure to check the optimize for web option!). If you have a few videos, [Zencoder](http://zencoder.com) is always a great choice :)
